### PR TITLE
feat(remix-react): add `RouteAnnouncer` component

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
Adding Route Announcer to `Remix js` for better accessibility.
Inspired by https://nextjs.org/docs/accessibility#route-announcements.

_Extremely sorry if I made any mistakes :(._

### CODE REVIEW

**Review by machour — CHANGES_REQUESTED** [review]
Hi @sanjaiyan-dev and thank you for this proposal

Could you target the `dev` branch instead of `main` ?

**Comment by sanjaiyan-dev** [comment]
> Hi @sanjaiyan-dev and thank you for this proposal
> 
> Could you target the `dev` branch instead of `main` ?

Okay 🙌

File: packages/remix-react/components.tsx [hunk 1]
```diff
@@ -227,6 +228,47 @@ function deserializeError(data: SerializedError): Error {
...
+        position: "absolute",
+        width: "1px",
+        whiteSpace: "nowrap",
+        wordWrap: "normal",
+      }}
+    >
+      {routeAnnouncement}
+    </p>
+  );
+});
```
Comment (MichaelDeBoey) [inline packages/remix-react/components.tsx:270]:
```suggestion
};
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: packages/remix-react/components.tsx [hunk 2]
```diff
@@ -227,6 +228,47 @@ function deserializeError(data: SerializedError): Error {
   return error;
 }
 
+const RouteAnnouncer = React.memo(() => {
```
[the remark this answers is not in the document]
Comment (sanjaiyan-dev) [reply packages/remix-react/components.tsx:231]: Sorry, I thought there might be a lot of state updates in the parents component which might lead to unnecessary re-render .so I wrap it with `React.memo` which does shallow comparison and prevent unnecessary re renders.

_Sorry if I made any mistakes :(_

**Comment by chaance** [comment]
This is something we had in a previous version of the `create-remix` template but ultimately removed, as there is quite a bit of nuance as to how this might work depending on a few factors—the main one being transition states. In some cases you may want to announce a pending transition, and we haven't come up with a solid API that abstracts the most common cases very well. It's something we're working on, but not quite ready to ship as this can already be handled better in user code.

**Pull request closed without merging**